tests/unit/loader/test_dataloader_backend.py

```diff
@@ -120,3 +122,33 @@ def test_dataloader_empty_error(datasets, engine, batch_size):
     assert "Neither Categorical or Continuous columns were found by the dataloader. " in str(
         exc_info.value
     )
+
+
+@pytest.mark.parametrize("engine", ["parquet"])
+@pytest.mark.parametrize("batch_size", [128])
+@pytest.mark.parametrize("epochs", [1, 5])
+def test_dataloader_epochs(datasets, engine, batch_size, epochs):
+    dataset = Dataset(str(datasets["parquet"]), engine=engine)
+    cont_names = ["x", "y", "id"]
+    cat_names = ["name-string", "name-cat"]
+    label_name = ["label"]
+
+    data_loader = DataLoader(
+        dataset,
+        cat_names=cat_names,
+        cont_names=cont_names,
+        batch_size=batch_size,
+        label_names=label_name,
+        shuffle=False,
+    )
+
+    # Convert to iterators and then to DataFrames
+    df1 = _concat(list(data_loader._buff.itr))
+    df2 = _concat(list(data_loader.epochs(epochs)._buff.itr))
```

Comment:
@jperez999 - Do you like this API any better? Note that `data_loader.epochs(n)` is used here to create a multi-epoch DataLoader object from the base (single-epoch) DataLoader.

Reply:
yeah this is better. It's allowed but not always set to a given number only during iteration.  I fell like I can get behind this.